### PR TITLE
Google Cloud Storage integration

### PR DESCRIPTION
**Pull request by jgagnon1**
Allow to use Google Cloud Storage as storage backend instead of S3.

### CODE REVIEW

File: src/main/java/com/pinterest/secor/common/SecorConfig.java [hunk 1]
```diff
@@ -255,6 +255,18 @@ public String getZookeeperPath() {
         return getString("secor.zookeeper.path");
     }
 
+    public String getGsCredentialsPath() {
+        return getString("secor.gs.credentials.path");
```
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/common/SecorConfig.java]: Can you add those params to one of the .properties file and document them?

File: src/main/java/com/pinterest/secor/uploader/GsUploadManager.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+     * globally shared instance across your application.
+     */
+    private static Storage mStorageService;
+
+    private Storage mClient;
+
+    public GsUploadManager(SecorConfig config) throws Exception {
+        super(config);
+
+        String credentialPath = System.getenv("GOOGLE_APPLICATION_CREDENTIALS");
```
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/uploader/GsUploadManager.java]: The priorities should be reversed: get the creds from the properties first and fall back to the env.

File: src/main/java/com/pinterest/secor/uploader/GsUploadManager.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+
+        return new FutureHandle(f);
+    }
+
+    private static Storage getService(String credentialsPath) throws Exception {
+        if (null == mStorageService) {
```
Comment (pgarbacki) [inline src/main/java/com/pinterest/secor/uploader/GsUploadManager.java]: if (mStorageService == null)

**Comment by pgarbacki** [comment]
@jgagnon1 Thanks for the contribution. Please address the comments and squash your commits.

File: src/main/java/com/pinterest/secor/common/SecorConfig.java (hunk 1 above)
Comment (jgagnon1) [inline src/main/java/com/pinterest/secor/common/SecorConfig.java]: Added in src/main/config/secor.dev.gs.partition.properties

**Comment by jgagnon1** [comment]
@pgarbacki Done and done; squashed

**Comment by jgagnon1** [comment]
Not sure to understand why the tests are failing after squashing?

**Comment by jgagnon1** [comment]
I rebased, but not sure what caused tests exceed 50 minutes. @pgarbacki ??

**Comment by pgarbacki** [comment]
@jgagnon1 :  I believe that this PR is at fault: https://github.com/pinterest/secor/pull/123

@kecaps are you looking into increasing the timeout?

**Comment by kecaps** [comment]
I've taken out the s3a testing which was doubling the time of integration tests. If you resubmit this to travis-ci, it should no longer timeout.

**Pull request merged**